more_itertools/more.py

```diff
@@ -2152,10 +2150,11 @@ class numeric_range(abc.Sequence, abc.Hashable):
 
     """
 
-    _EMPTY_HASH = hash(range(0, 0))
+    _EMPTY_HASH = hash(range(0))
 
     def __init__(self, *args):
         argc = len(args)
+        cls_name = self.__class__.__name__
```

Comment:
Dragged from the #934 .

Reply:
This group of edits is entirely unnecessary and do not improve the user experience.

It makes instances a little larger and slows instantiation.

If Bo does want to factor-out the class name to make this easily more subclassable (maybe he does, maybe he doesn't), the usual way to get the class name with `to compute `{type(self).__name__}` in the relevant f-strings.